### PR TITLE
Feature/81-routelist

### PR DESCRIPTION
**Pull request by adCBueno**

### CODE REVIEW

File: RoutesApp_ios/SceneDelegate.swift [hunk 1]
```diff
@@ -67,6 +67,7 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
         tabBarVC.tabBar.backgroundColor = .white
         // Add VCs to TabBarVC
         tabBarVC.viewControllers = [
+            createNavController(for: RouteListViewController(), title: "Home", image: UIImage(systemName: "bus.fill")!),
```
Comment (IrvingJVR-MX) [inline RoutesApp_ios/SceneDelegate.swift]: i think this should be "String.localizeString(localizedString: "home") "   :)

File: RoutesApp_ios/modules/routelist/viewmodels/RouteDetailViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    
+    func getLines(completion: @escaping () -> ()) {
+        
+        firebaseManager.getLines(type: Lines.self, forCollection: .Lines) { result in
+            switch result {
+            case .success(let lines):
+                self.routeListModel = lines
+                self.getCategories(completion: completion)
+            case .failure(let error):
+                print(error)
```
Comment (IrvingJVR-MX) [inline RoutesApp_ios/modules/routelist/viewmodels/RouteDetailViewModel.swift]: i think this print shouldn´t go

File: RoutesApp_ios/modules/routelist/viewmodels/RouteDetailViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    func getCategories(completion: @escaping () -> ()) {
+        firebaseManager.getLinesCategory(type: LinesCategory.self, forCollection: .Lines) { result in
+            switch result {
+            case .success(let lines):
+                self.linesCategory = lines
+                self.mapRouteListDetailModel()
+                completion()
+            case .failure(let error):
+                print(error)
```
Comment (IrvingJVR-MX) [inline RoutesApp_ios/modules/routelist/viewmodels/RouteDetailViewModel.swift]: also here

File: RoutesApp_ios/modules/routelist/views/controllers/RouteListViewController.swift [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupNavigationBar()
+        setIcon()
+        routeListDetailViewModel.getLines {
+            self.routeListTableView.reloadData()
+        }
+
+        routeListTableView.register(UINib.init(nibName: "RouteListTableViewCell", bundle: nil), forCellReuseIdentifier: "RouteListTableViewCell")
```
Comment (jmacboy) [inline RoutesApp_ios/modules/routelist/views/controllers/RouteListViewController.swift]: We're using nibNames and reuseIdentifiers as constants, add those constants in the constants class and call them here

File: RoutesApp_ios/modules/routelist/views/controllers/RouteListViewController.swift [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        self.routeListTableView.reloadData()
+    }
+    
+    func setupNavigationBar() {
+        navigationItem.title = "Route List"
```
Comment (jmacboy) [inline RoutesApp_ios/modules/routelist/views/controllers/RouteListViewController.swift]: this text should be localized

File: RoutesApp_ios/modules/routelist/views/controllers/RouteListViewController.swift [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    @objc func showFilterPopUp() {
+        
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return routeListDetailViewModel.routeListDetailModels.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let tableViewCell = tableView.dequeueReusableCell(withIdentifier: "RouteListTableViewCell", for: indexPath) as? RouteListTableViewCell else {
```
Comment (jmacboy) [inline RoutesApp_ios/modules/routelist/views/controllers/RouteListViewController.swift]: same here, use a constant for the cell identifier

**Review by jmacboy — CHANGES_REQUESTED** [review]

File: RoutesApp_ios/modules/routelist/views/controllers/RouteListViewController.swift (hunk 4 above)
Comment (adCBueno) [reply RoutesApp_ios/modules/routelist/views/controllers/RouteListViewController.swift]: Changed :)

File: RoutesApp_ios/modules/routelist/views/controllers/RouteListViewController.swift (hunk 5 above)
Comment (adCBueno) [reply RoutesApp_ios/modules/routelist/views/controllers/RouteListViewController.swift]: Changed :)

File: RoutesApp_ios/modules/routelist/views/controllers/RouteListViewController.swift (hunk 6 above)
Comment (adCBueno) [reply RoutesApp_ios/modules/routelist/views/controllers/RouteListViewController.swift]: Changed :)

**Review by jmacboy — APPROVED** [review]

**Review by IrvingJVR-MX — APPROVED** [review]

**Pull request merged**